pgmpy/models/MarkovChain.py

```diff
@@ -0,0 +1,299 @@
+        """
+        Checks if the a dict representation of a states of the variables is valid.
+        """
+        if not isinstance(state, dict):
+            raise ValueError('Start state must be a dict.')
+        if not set(state.keys()) == set(self.transition_models.keys()):
+            raise ValueError('Start state must represent a complete assignment to all variables.')
+        for var, val in state.items():
+            if val >= self.cardinalities[var]:
+                raise ValueError('Assignment {val} to {var} invalid.'.format(val=val, var=var))
+        return True
+
+    def add_variable(self, variable, card=0):
+        """
+        Add a variable to the model.
+
+        Parameters:
+        -----------
+        variable: any hashable python object
+        card: int
+            representing the cardinality of the variable to be added.
+
+        Examples:
+        ---------
+        >>> from pgmpy.models import MarkovChain as MC
+        >>> model = MC()
+        >>> model.add_variable('x', 4)
+        """
+        assert variable not in self.variables
+        self.variables.append(variable)
+        self.cardinalities[variable] = card
+        self.transition_models[variable] = {}
+
+    def add_variables_from(self, variables, cards):
+        """
+        Add several variables to the model at once.
+
+        Parameters:
+        -----------
+        variables: list
+            list of variables to be added.
+        cards: list
+            list of cardinalities of the variables to be added.
+
+        Examples:
+        ---------
+        >>> from from pgmpy.models import MarkovChain as MC
+        >>> model = MC()
+        >>> model.add_variables_from(['x', 'y'], [3, 4])
+        """
+        for var, card in zip(variables, cards):
+            self.add_variable(var, card)
+
+    def add_transition_model(self, variable, transition_model):
+        """
+        Adds a transition model for a particular variable.
+
+        Parameters:
+        -----------
+        variable: any hashable python object
+            must be an existing variable of the model.
+        transition_model: dict
+            representing valid transition probabilities defined for every possible state of the variable.
+
+        Examples:
+        ---------
+        >>> from pgmpy.models import MarkovChain as MC
+        >>> model = MC()
+        >>> model.add_variable('grade', 3)
+        >>> grade_tm = {0: {0: 0.1, 1: 0.5, 2: 0.4}, 1: {0: 0.2, 1: 0.2, 2: 0.6 }, 2: {0: 0.7, 1: 0.15, 2: 0.15}}
+        >>> model.add_transition_model('grade', grade_tm)
+        """
+        # check if the transition model is valid
+        if not isinstance(transition_model, dict):
+            raise ValueError('Transition model must be a dict.')
+        if not set(transition_model.keys()) == set(range(self.cardinalities[variable])):
+            raise ValueError('Transitions must be defined for all states of variable {var}.'.format(var=variable))
```

Comment:
@pratyakshs You can also add the expected states and and states got in the error message. Will make debugging more easier.